src/giftCards/GiftCardUpdate/providers/GiftCardUpdateDialogsProvider/GiftCardUpdateDialogsProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import { giftCardsListPath, giftCardUrl } from "@saleor/giftCards/urls";
+import useNavigator from "@saleor/hooks/useNavigator";
+import createDialogActionHandlers from "@saleor/utils/handlers/dialogActionHandlers";
+import React, { createContext } from "react";
+
+import {
+  GiftCardUpdatePageActionParamsEnum,
+  GiftCardUpdatePageUrlQueryParams
+} from "../../types";
+
+interface GiftCardUpdateDialogsProviderProps {
+  children: React.ReactNode;
+  params: GiftCardUpdatePageUrlQueryParams;
+  id: string;
+}
+
+export interface GiftCardUpdateDialogsConsumerProps {
+  navigateBack: () => void;
+  openSetBalanceDialog: () => void;
+  isSetBalanceDialogOpen: boolean;
+  closeDialog: () => void;
+}
+
+export const GiftCardUpdateDialogsContext = createContext<
+  GiftCardUpdateDialogsConsumerProps
+>(null);
+
+const GiftCardUpdateDialogsProvider: React.FC<GiftCardUpdateDialogsProviderProps> = ({
+  children,
+  params,
+  id
+}) => {
+  const navigate = useNavigator();
+
+  const [openDialog, closeDialog] = createDialogActionHandlers<
+    GiftCardUpdatePageActionParamsEnum,
+    GiftCardUpdatePageUrlQueryParams
+  >(navigate, params => giftCardUrl(id, params), params);
+
+  const openSetBalanceDialog = () =>
+    openDialog(GiftCardUpdatePageActionParamsEnum.SET_BALANCE);
```

Comment:
Where is this Dialog defined (as a component)?

Reply:
`src/giftCards/GiftCardUpdate/GiftCardUpdateBalanceDialog/GiftCardUpdateBalanceDialog.tsx`